script.js

```diff
@@ -1,49 +1,85 @@
-const db_local = 'stays.json'
-const db_url = 'https://raw.githubusercontent.com/pamdayne/wind-d-bnb/main/db/stays.json'
+const DB_LOCAL = "stays.json";
+const DB_URL =
+  "https://raw.githubusercontent.com/pamdayne/wind-d-bnb/main/db/stays.json";
 
 async function fetchLocations() {
-	try {
-		let resp = await fetch(db_url)
-		return await resp.json()
-	} catch (err) {
-		console.error(err)
-	}
+  return fetch(DB_URL)
+    .then(function(resp){return resp.json()})
+    .then(function(json){
+       return json;
+    }).catch(function(error){
+      console.error(error);
+    })
 }
 
 async function renderList() {
-	let places = await fetchLocations()
-	let html = ''
-
-	places.forEach(place => {
-		setCountry(place.country)
-
-		html += `<div class="lists-wrapper">
-								<div class="photo">
-									<img
-										src="${place.photo}"
-										alt="${place.title}" />
-								</div>
-								<div class="details">
-									<div class="unit">
-									${place.superHost != false ? '<div class="host"> Super Host</div>' : ''}
-										<div class="type">${place.type}</div >
-										<div class="bed">
-										${place.beds != null ? '<span class="total">' + place.beds + ' beds</span > ' : ''}
-										</div >
-										<div class="score">
-										<span class="material-icons">star</span>
-										<span class="rating">${place.rating}</span>
-										</div>
-									</div>
-									<div class="short-desc">${place.title}</div>
-								</div >
-							</div > `
-		document.querySelector('#locations').innerHTML = html
-	})
+  let places = await fetchLocations()
+
+  // HOW CAN WE WRITE THIS BETTER?
+  // Is this the better way of doing it?
+  let featImg, typeDiv, hostDiv, rateDiv, bedDiv, descDiv, unitDiv, detailsDiv, mainDiv = {}
+
+  places.forEach(location => {
+    const { photo, title, superHost, type, beds, rating} = location;
+
+    // The place's image
+    featImg = setFeaturedImage(photo, title, 'photo')
+
+    // Unit's details
+    hostDiv = superHost ? setContentElement('div', 'SUPER HOST', 'host') : ''
+    typeDiv = setContentElement('div', type, 'type')
+    bedDiv = beds ? setContentElement('div', beds, 'bed') : ''
+    rateDiv = setContentElement('div', rating, 'rating')
+    rateDiv.insertBefore(setIconElement('star'), rateDiv.firstChild)
+
+    // Wraps the details
+    unitDiv = setContentElement('div', '', 'unit')
+    unitDiv.append(hostDiv, typeDiv, bedDiv, rateDiv)
+
+    descDiv = setContentElement('div', title, 'short-desc')
+
+    detailsDiv = setContentElement('div', '', 'details')
+    detailsDiv.append(unitDiv, descDiv)
+
+    mainDiv = setContentElement('div', '', 'list-wrapper')
+    mainDiv.append(featImg, detailsDiv)
+
+    // IMPROVE: #locations.innerHTML will get replaced with the html in every iteration
+    document.querySelector('#locations').appendChild(mainDiv)
```

Comment:
I'm still figuring out this one. 😓

Reply:
shift the `appendChild` outside the `forEach` let the loop finish the append the whole chunk

Reply:
Let me try to do this!
Will also try to equal the whole html into a variable. then maybe only call this.